src/test/fuzz/eval_script.cpp

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2009-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <script/interpreter.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+
+#include <limits>
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    if (buffer.size() <= sizeof(unsigned int)) {
+        return;
+    }
```

Comment:
why is this needed? The fuzzer should be able to learn it trivially.

Also, you are limiting the script to have a size greater or equal to one

Reply:
Unfortunately needed to not trigger the following UBSan warnings in Chromium's `FuzzedDataProvider.h` :)

Without:

```
$ src/test/fuzz/eval_script
INFO: Seed: 2056324753
INFO: Loaded 1 modules   (453398 inline 8-bit counters): 453398 [0x563c78576048, 0x563c785e4b5e),
INFO: Loaded 1 PC tables (453398 PCs): 453398 [0x563c785e4b60,0x563c78ccfcc0),
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
test/fuzz/FuzzedDataProvider.h:212:17: runtime error: null pointer passed as argument 1, which is declared to never be null
/usr/include/string.h:43:28: note: nonnull attribute specified here
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior test/fuzz/FuzzedDataProvider.h:212:17 in
test/fuzz/FuzzedDataProvider.h:212:32: runtime error: null pointer passed as argument 2, which is declared to never be null
/usr/include/string.h:43:28: note: nonnull attribute specified here
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior test/fuzz/FuzzedDataProvider.h:212:32 in
```

Reply:
Why doesn't that make the fuzzer fail and exit?

Reply:
Anyway, the check should be `<`, not `<=`, otherwise you are limiting the script to have a size greater or equal to one

Reply:
@MarcoFalke I think you forgot to set `UBSAN_OPTIONS=print_stacktrace=1:halt_on_error=1:exitcode=1` :)

I think this is the behaviour you expected:

```
$ UBSAN_OPTIONS=print_stacktrace=1:halt_on_error=1:exitcode=1 src/test/fuzz/eval_script -runs=1 ../btc_qa_assets/fuzz_seed_corpus/eval_script
INFO: Seed: 3606842015
INFO: Loaded 1 modules   (453402 inline 8-bit counters): 453402 [0x55b0f0056068, 0x55b0f00c4b82),
INFO: Loaded 1 PC tables (453402 PCs): 453402 [0x55b0f00c4b88,0x55b0f07afd28),
INFO:        0 files found in ../btc_qa_assets/fuzz_seed_corpus/eval_script
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
test/fuzz/FuzzedDataProvider.h:212:17: runtime error: null pointer passed as argument 1, which is declared to never be null
/usr/include/string.h:43:28: note: nonnull attribute specified here
    #0 0x55b0eb940b1d in std::vector<unsigned char, std::allocator<unsigned char> > FuzzedDataProvider::ConsumeBytes<unsigned char>(unsigned long, unsigned long) /src/./test/fuzz/FuzzedDataProvider.h:212:5
    #1 0x55b0eb9404b7 in std::vector<unsigned char, std::allocator<unsigned char> > FuzzedDataProvider::ConsumeBytes<unsigned char>(unsigned long) /src/./test/fuzz/FuzzedDataProvider.h:42:12
    #2 0x55b0eb93e4c3 in std::vector<unsigned char, std::allocator<unsigned char> > FuzzedDataProvider::ConsumeRemainingBytes<unsigned char>() /src/./test/fuzz/FuzzedDataProvider.h:143:12
    #3 0x55b0eb93d70f in test_one_input(std::vector<unsigned char, std::allocator<unsigned char> > const&) /src/test/fuzz/eval_script.cpp:18:68
    #4 0x55b0eb9331fb in LLVMFuzzerTestOneInput /src/test/fuzz/fuzz.cpp:34:5
    #5 0x55b0eb6fbb87 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (/src/test/fuzz/eval_script+0x1da8b87)
    #6 0x55b0eb7057cb in fuzzer::Fuzzer::ReadAndExecuteSeedCorpora(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, fuzzer::fuzzer_allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) (/src/test/fuzz/eval_script+0x1db27cb)
    #7 0x55b0eb707932 in fuzzer::Fuzzer::Loop(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, fuzzer::fuzzer_allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) (/src/test/fuzz/eval_script+0x1db4932)
    #8 0x55b0eb6f6e1c in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) (/src/test/fuzz/eval_script+0x1da3e1c)
    #9 0x55b0eb6bb2c2 in main (/src/test/fuzz/eval_script+0x1d682c2)
    #10 0x7f84c2817b96 in __libc_start_main /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310
    #11 0x55b0eb6e9d39 in _start (/src/test/fuzz/eval_script+0x1d96d39)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior test/fuzz/FuzzedDataProvider.h:212:17 in
MS: 0 ; base unit: 0000000000000000000000000000000000000000


artifact_prefix='./'; Test unit written to ./crash-da39a3ee5e6b4b0d3255bfef95601890afd80709
Base64:
$ echo $?
1
```